### PR TITLE
[DNM] rgw: remove buffering from the ASIO and CivetWeb frontends

### PR DESCRIPTION
**Pull request by rzarzynski**

### CODE REVIEW

**Comment by cbodley** [comment]
@rzarzynski isn't this the buffering that was added to batch up a bunch of small writes? or is that limited to the `rgw::io::BuffererSink` and `rgw::io::StaticOutputBufferer<>` pieces?

**Comment by rzarzynski** [comment]
@cbodley: nope, it's a different thing. `rgw::io::StaticOutputBufferer` is intended not for calculating the content length, but to avoid going to kernel with huge number of small chunks during the HTTP header composition phase.

**Comment by rzarzynski** [comment]
@cbodley: it looks we have a problem here. Upper layers (formatters, `*::send_response` methods?) tend to issue a lot of small chunks. For instance, each item of the account listing is flushed separately. :-( I'm afraid we're going to have similar degradation like with the composition of HTTP header. Tomorrow I will paste a dump from `ngrep` to illustrate the fragmentation.

**Comment by rzarzynski** [comment]
That's the dump (CivetWeb):
```
$ sudo ngrep -W byline port 8000 -d lo
interface: lo (127.0.0.0/255.0.0.0)
filter: (ip or ip6) and ( port 8000 )

####
T 127.0.0.1:33825 -> 127.0.0.1:8000 [AP]
GET /v1/KEY_2987047fc47840058e89e6182e0d96c3/ HTTP/1.1.
User-Agent: curl/7.35.0.
Host: 127.0.0.1:8000.
Accept: */*.
X-Auth-Token: daeebbd3127a4a5195a61c25dce10b1f.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
HTTP/1.1 200 OK.
X-Timestamp: 1488213326.66676.
X-Account-Container-Count: 6.
X-Account-Object-Count: 0.
X-Account-Bytes-Used: 0.
X-Account-Bytes-Used-Actual: 0.
X-Account-Access-Control: {}.
X-Trans-Id: tx000000000000000000003-0058b4554e-101c-default.
Content-Type: text/plain; charset=utf-8.
Transfer-Encoding: chunked.
Date: Mon, 27 Feb 2017 16:35:26 GMT.
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
c.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
a--776439599
##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
d.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]

b--460333876
##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
d.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]

c--561461791
##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
e.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]

d--1678161372
##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
e.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]

e--1735901550
##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
d.

##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]

f--510976939
##
T 127.0.0.1:8000 -> 127.0.0.1:33825 [AP]
.
0.
.

####^Cexit
92 received, 0 dropped
```

File: src/rgw/rgw_asio_frontend.cc [hunk 1]
```diff
@@ -94,10 +94,9 @@ class AsioConnection : public std::enable_shared_from_this<AsioConnection> {
     RGWRequest req{env.store->get_new_req_id()};
     RGWAsioClientIO real_client{std::move(socket), std::move(request)};
     auto real_client_io = rgw::io::add_reordering(
-                            rgw::io::add_buffering(
-                              rgw::io::add_chunking(
-                                rgw::io::add_conlen_controlling(
-                                  &real_client))));
+                            rgw::io::add_chunking(
```
Comment (amitkumar50) [inline src/rgw/rgw_asio_frontend.cc]: @rzarzynski (NIT) Please keep title of PR <50 characters.
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

**Comment by cbodley** [comment]
hi @rzarzynski, it sounds like we concluded that this approach wouldn't work - okay to close the pr?

**Comment by rzarzynski** [comment]
Sure, closing.

**Pull request closed without merging**